LevelSpaceXWGUI.nlogo

```diff
@@ -699,27 +702,30 @@ to-report get-eligible-interactions [an-entity]
   if the-type = "observer"[
     report map [first ?] filter [
       table:get last ? "type" = "command" and 
-      member? "O" table:get last ? "contexts" 
+      member? "O" table:get last ? "contexts"
+      and table:get last ? "visible"
     ] 
     table:to-list tasks 
   ]  
   ;; if it's an agentset, they can call turtle commands in their own model or observer commands in other models
   if the-type = "agentset"[
     report map [first ?] filter [
-      (table:get last ? "type" = "command" and 
+      ((table:get last ? "type" = "command" and 
         member? "T" table:get last ? "contexts" and
         table:get last ? "model" = table:get an-entity "model"
         )
       or
       (table:get last ? "type" = "command" and 
         member? "O" table:get last ? "contexts"  and     
-        table:get last ? "model" != table:get an-entity "model")
+        table:get last ? "model" != table:get an-entity "model"))
+      and table:get last ? "visible"
     ] 
     table:to-list tasks 
   ]
   if the-type = "value"[
     report filter [
-      table:get last ? "type" = "value"
+      table:get last ? "type" = "value" and
+      table:get last ? "visible"
```

Comment:
There's a lot of shared logic in these cases that could probably be pulled out into a reporter. Something like:
```netlogo
to-report visible-interaction-filter [t type context other-filter]
  task [
    table:get last ? "type" = type and
    table:get last ? "visible" and
    run-result other-filter (table-get last ?)
  ]
end
```